inference-engine/src/transformations/src/transformations/op_conversions/convert_slice_to_strided_slice.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include "transformations/op_conversions/convert_slice_to_strided_slice.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/validation_util.hpp"
+
+#include "itt.hpp"
+
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::SliceToStridedSlice, "SliceToStridedSlice", 0);
+
+namespace {
+    Output<ngraph::Node> align_indices(const Output<ngraph::Node>& indices,
+                                   const Output<ngraph::Node>& slice_axes,
+                                   const Output<ngraph::Node>& scatter_axis,
+                                   size_t slice_indices_length,
+                                   int64_t fill_in_value,
+                                   NodeVector& new_ops) {
+    // Handle a case when starts/ends/steps lengths are less than provided axes
+    // in order to ensure compatibility with `StridedSlice:v1` interface
+    // Example:
+    // data_shape: {3, 3, 3, 3}
+    // starts: [1, 1] - after extending --> [0, 0, 1, 1]
+    // ends: [2, 2] - after extending --> [0, 0, 2, 2]
+    // steps : [1, 1] - after extending --> [1, 1, 1, 1]
+    // axes: [2, 3] - apply slice values to 2 and 3 dimension of input data
+    // expected_output_shape: {3, 3, 1, 1}
+
+    const auto default_indices = ngraph::opset8::Constant::create(indices.get_element_type(), Shape{slice_indices_length}, {fill_in_value});
+    std::shared_ptr<ngraph::Node> adjusted_indices = std::make_shared<ngraph::opset8::ScatterUpdate>(default_indices,
```

Comment:
As we have already discussed, 'indices' input here is not always a Constant, so ScatterUpdate won't be const folded here and will be passed to the plugins for execution as is. So we need to check ScatterUpdate operation coverage in the plugins, supported this op or not and compare to Gather op coverage.
@GlebKazantaev mentioned that @andrei-kochin might have a table with operations coverage for the plugins, where each operation is actually supported. @andrei-kochin could you share this table with @mitruska?

Reply:
Conslusion of another discussion is:
If any of Slice/StridedSlice indices input is a Parameter, we always get dynamic shape 
(not supported by plugins before, so any implementation is backward compatible),
and for Constant inputs any Op used for the alignment will be can be constant folded.